roles/container-engine/docker/vars/kylin.yml

```diff
@@ -1,54 +1,61 @@
-  '1.6.13': "{{ containerd_package }}-1.6.13-3.1.el{{ ansible_distribution_major_version }}"
-  '1.6.14': "{{ containerd_package }}-1.6.14-3.1.el{{ ansible_distribution_major_version }}"
-  '1.6.15': "{{ containerd_package }}-1.6.15-3.1.el{{ ansible_distribution_major_version }}"
-  '1.6.16': "{{ containerd_package }}-1.6.16-3.1.el{{ ansible_distribution_major_version }}"
-  '1.6.18': "{{ containerd_package }}-1.6.18-3.1.el{{ ansible_distribution_major_version }}"
-  '1.6.28': "{{ containerd_package }}-1.6.28-3.1.el{{ ansible_distribution_major_version }}"
-  'stable': "{{ containerd_package }}-1.6.28-3.1.el{{ ansible_distribution_major_version }}"
-  'edge': "{{ containerd_package }}-1.6.28-3.1.el{{ ansible_distribution_major_version }}"
-
-docker_version: 19.03
-docker_cli_version: 19.03
+  '1.3.7': "{{ containerd_package }}-1.3.7-3.1.el8"
+  '1.3.9': "{{ containerd_package }}-1.3.9-3.1.el8"
+  '1.4.3': "{{ containerd_package }}-1.4.3-3.2.el8"
+  '1.4.4': "{{ containerd_package }}-1.4.4-3.1.el8"
+  '1.4.6': "{{ containerd_package }}-1.4.6-3.1.el8"
+  '1.4.9': "{{ containerd_package }}-1.4.9-3.1.el8"
+  '1.4.12': "{{ containerd_package }}-1.4.12-3.1.el8"
+  '1.6.4': "{{ containerd_package }}-1.6.4-3.1.el8"
+  '1.6.6': "{{ containerd_package }}-1.6.6-3.1.el8"
+  '1.6.7': "{{ containerd_package }}-1.6.7-3.1.el8"
+  '1.6.8': "{{ containerd_package }}-1.6.8-3.1.el8"
+  '1.6.9': "{{ containerd_package }}-1.6.9-3.1.el8"
+  '1.6.10': "{{ containerd_package }}-1.6.10-3.1.el8"
+  '1.6.11': "{{ containerd_package }}-1.6.11-3.1.el8"
+  '1.6.12': "{{ containerd_package }}-1.6.12-3.1.el8"
+  '1.6.13': "{{ containerd_package }}-1.6.13-3.1.el8"
+  '1.6.14': "{{ containerd_package }}-1.6.14-3.1.el8"
+  '1.6.15': "{{ containerd_package }}-1.6.15-3.1.el8"
+  '1.6.16': "{{ containerd_package }}-1.6.16-3.1.el8"
+  '1.6.18': "{{ containerd_package }}-1.6.18-3.1.el8"
+  '1.6.28': "{{ containerd_package }}-1.6.28-3.1.el8"
+  '1.6.31': "{{ containerd_package }}-1.6.31-3.1.el8"
+  'stable': "{{ containerd_package }}-1.6.31-3.1.el8"
+  'edge': "{{ containerd_package }}-1.6.31-3.1.el8"
 
 # https://docs.docker.com/engine/installation/linux/centos/#install-from-a-package
-# https://download.docker.com/linux/centos/<centos_version>>/x86_64/stable/Packages/
+# https://download.docker.com/linux/centos/8/x86_64/stable/Packages/
 # or do 'yum --showduplicates list docker-engine'
 docker_versioned_pkg:
   'latest': docker-ce
-  '18.09': docker-ce-3:18.09.9-3.el{{ ansible_distribution_major_version }}
-  '19.03': docker-ce-3:19.03.15-3.el{{ ansible_distribution_major_version }}
-  '23.0': docker-ce-3:23.0.6-1.el{{ ansible_distribution_major_version }}
-  '24.0': docker-ce-3:24.0.9-1.el{{ ansible_distribution_major_version }}
-  'stable': docker-ce-3:24.0.9-1.el{{ ansible_distribution_major_version }}
-  'edge': docker-ce-3:24.0.9-1.el{{ ansible_distribution_major_version }}
+  '18.09': docker-ce-3:18.09.9-3.el8
```

Comment:
Why do we don't use ansible_distribution_major_version anymore here?

Reply:
😄 Currently, in the Kylin v10 environment, only Docker RPM packages compatible with CentOS 8 can be used. The previous approach involved using set_fact to set ansible_distribution_major_version to 8 to achieve the desired result. However, this method is not considered best practice and has been deprecated. Therefore, the latest approach is to directly specify the package names related to CentOS 8.

Reply:
I see, thanks for your details!

/lgtm